src/any_agent/tracing/agent_trace.py

```diff
@@ -82,6 +83,69 @@ def compute_cost_info(
     return CostInfo.model_validate(new_info)
 
 
+def spans_to_messages(spans: list[AgentSpan]) -> list[dict[str, str]]:
+    """Convert spans to standard message format.
+
+    Args:
+        spans: List of AgentSpan objects to convert.
+
+    Returns:
+        List of message dicts with 'role' and 'content' keys.
+
+    """
+    messages: list[dict[str, str]] = []
+
+    # Process spans in chronological order (excluding the final invoke_agent span)
+    # Filter out any non-AgentSpan objects and agent invocation spans
+    filtered_spans: list[AgentSpan] = []
+    for span in spans:
+        if hasattr(span, "is_agent_invocation") and not span.is_agent_invocation():
```

Comment:
How can there be an span that is not an `AgentSpan`?

Reply:
Just in case someone was calling the method with a badly constructed list. But, if we move this to be a method of AgentTrace, the corner case goes away 👍